packages/expo-cli/src/commands/run/android/runAndroid.ts

```diff
@@ -146,6 +153,38 @@ export async function runAndroidActionAsync(projectRoot: string, options: Option
   }
 }
 
+function track(projectRoot: string, exp: ExpoConfig) {
+  UnifiedAnalytics.logEvent('dev client run command', {
+    status: 'started',
+    platform: 'android',
+    ...getDevClientProperties(projectRoot, exp),
+  });
+  StatusEventEmitter.once('bundleBuildFinish', () => {
+    // Send the 'bundle ready' event once the JS has been built.
+    UnifiedAnalytics.logEvent('dev client run command', {
+      status: 'bundle ready',
+      platform: 'android',
+      ...getDevClientProperties(projectRoot, exp),
+    });
+  });
+  StatusEventEmitter.once('deviceLogReceive', () => {
+    // Send the 'ready' event once the app is running in a device.
+    UnifiedAnalytics.logEvent('dev client run command', {
+      status: 'ready',
+      platform: 'android',
+      ...getDevClientProperties(projectRoot, exp),
+    });
+  });
+  installCustomExitHook(() => {
+    UnifiedAnalytics.logEvent('dev client run command', {
+      status: 'finished',
+      platform: 'android',
+      ...getDevClientProperties(projectRoot, exp),
+    });
+    UnifiedAnalytics.flush();
```

Comment:
does this [flush](https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/exp.ts#L354) not catch this scenario?

Reply:
I think it won't handle the case where an error is thrown. Perhaps it should be moved to a `finally` statement.

Reply:
thats a good point. maybe something for a subsequent pr then

Reply:
Tried to use `finally`, however it turns out it won't work for the error case like that. There's a `process.exit(1)` in the `catch` block which means the `finally` block won't execute. We would have to add `flush()` to the `catch` block instead.

However, there's another reason the default [flush](https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/exp.ts#L354) code won't help with these events. For commands like `expo start` or `expo run` that start the development server, that flush will happen as soon as the command action code has executed – i.e. as soon as the server has started. But I need this flush to happen when the server is shutting down and after logging the event.

If we add other events to trigger during shutdown in the future, we could see if it makes sense to add this to the default exit hook or something similar (however since the order of exit event handlers is not guaranteed, we'd have to make some changes for this to work). But the ones I added are the only shutdown events tracked at the moment, as far as I know, so I'm going to keep the code as it is for now.